app/controllers/api/posts_controller.rb

```diff
@@ -3,7 +3,27 @@ def create
     post = Post.new post_params
 
     if post.save
-      render json: :nothing, status: 200
+      render json: { id: post.id }, status: 200
+    else
+      render json: { errors: post.errors.full_messages }.to_json, status: 422
+    end
+  end
+
+  def update
+    post = Post.find(params[:id])
+
+    if post.published?
+      render json: 'Already published'.to_json, status: 422
```

Comment:
Is this really intended? I often find it useful to edit published posts, usually to fix typos. BUt maybe we only want to do that via the blog interface?